chromadb/test/test_api.py

```diff
@@ -1368,3 +1368,22 @@ def test_multiple_collections(api_fixture, request):
 
     assert results1["ids"][0][0] == ids1[0]
     assert results2["ids"][0][0] == ids2[0]
+
+    
+@pytest.mark.parametrize("api_fixture", test_apis)
+def test_upsert(api_fixture, request):
+    api = request.getfixturevalue(api_fixture.__name__)
+    api.reset()
+    collection = api.create_collection("test")
+    
+    # Add some items via upsert
+    collection.upsert(ids=["id1", "id2", "id3"], documents=["hello", "world", "foo"])
+    assert collection.count() == 3
+
+    # Add an item with the same ID 
+    collection.upsert(ids=["id1", "id4"], documents=["bar", "baz"])
+
+    # We should expect there to be only one item, the "bar" one
+    items = collection.get(ids="id1")
+    assert len(items["ids"]) == 1
```

Comment:
We should add assertions that the total collection size is 4, and that the new embedding is reachable via KNN search. That's an important part of the write path that's not tested yet.